linter/pwnedkeys/handler.go

```diff
@@ -0,0 +1,113 @@
+package pwnedkeys
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/pkimetal/pkimetal/config"
+	"github.com/pkimetal/pkimetal/linter"
+)
+
+type Pwnedkeys struct{}
+
+var httpClient http.Client
+
+func init() {
+	httpClient = http.Client{
+		Timeout: config.Config.Linter.Pwnedkeys.HTTPTimeout,
+	}
+
+	// Register pwnedkeys.
+	(&linter.Linter{
+		Name:         "pwnedkeys",
+		Version:      "1.0.0",
+		Url:          "https://pwnedkeys.com/api/v1",
+		Unsupported:  linter.NonCertificateProfileIDs,
+		NumInstances: config.Config.Linter.Pwnedkeys.NumGoroutines,
+		Interface:    func() linter.LinterInterface { return &Pwnedkeys{} },
+	}).Register()
+}
+
+func (l *Pwnedkeys) StartInstance() (useHandleRequest bool, directory, cmd string, args []string) {
+	return true, "", "", nil // The pwnedkeys API is called from Goroutine(s) in the pkimetal process, so there are no "external" instances.
+}
+
+func (l *Pwnedkeys) StopInstance(lin *linter.LinterInstance) {
+}
+
+func (l *Pwnedkeys) HandleRequest(lin *linter.LinterInstance, lreq *linter.LintingRequest, ctx context.Context) []linter.LintingResult {
+	var lres []linter.LintingResult
+	var httpRequest *http.Request
+	var err error
+	s := sha256.Sum256(lreq.Cert.RawSubjectPublicKeyInfo)
+	if httpRequest, err = http.NewRequest(http.MethodGet, fmt.Sprintf("https://v1.pwnedkeys.com/%s", hex.EncodeToString(s[:])), nil); err != nil {
+		lres = append(lres, linter.LintingResult{
+			Severity: linter.Severity[config.Config.Linter.Pwnedkeys.APIErrorSeverity],
+			Finding:  err.Error(),
+		})
+	} else {
+		httpRequest.Header.Set("User-Agent", linter.PKIMETAL_NAME)
```

Comment:
Rather than set these here, you can set a Transport for your http Client that you create here that adds these for you

Reply:
Addressed by https://github.com/pkimetal/pkimetal/pull/183/commits/64025820efd14b886bf88be76a5b3023a8903e5d.